kratos/includes/parallel_environment.h

```diff
@@ -105,6 +112,17 @@ class KRATOS_API(KRATOS_CORE) ParallelEnvironment
      */
     static FillCommunicator::Pointer CreateFillCommunicator(ModelPart& rModelPart);
 
+    /**
+     * @brief Create a Communicator object
+     * This method uses the previously registered communicator factory for the creation of a new communicator pointer
+     * @param rModelPart Model part required to retrieve the variables list from it
+     * @param rDataCommunicatorName Name of the data communicator to be retrieved for the communicator construction
+     * @return Communicator::UniquePointer Unique pointer to the new communicator
+     */
+    static Communicator::UniquePointer CreateCommunicator(
+      ModelPart& rModelPart,
+      const std::string& rDataCommunicatorName);
```

Comment:
I would add a second fct that directly takes the `DataCommunicator`, for the sake of completeness

Reply:
Indeed, this was the original design (I removed it in favor of the string-based one). Makes sense to keep both. :ok: